velox/functions/prestosql/URLFunctions.h

```diff
@@ -30,10 +31,10 @@ FOLLY_ALWAYS_INLINE StringView submatch(const boost::cmatch& match, int idx) {
 template <typename TInString>
 bool parse(const TInString& rawUrl, boost::cmatch& match) {
   static const boost::regex kUriRegex(
-      "([a-zA-Z][a-zA-Z0-9+.-]*):" // scheme:
-      "([^?#]*)" // authority and path
-      "(?:\\?([^#]*))?" // ?query
-      "(?:#(.*))?"); // #fragment
+      "^(([^:\\/?#]+):)?" // scheme:
```

Comment:
These regexes are not very readable. Maybe add comments to explain each of them?